src/telegrafs/components/TelegrafConfigOverlay.tsx

```diff
@@ -1,109 +1,55 @@
 // Libraries
-import React, {PureComponent} from 'react'
-import {connect} from 'react-redux'
+import React, {FC} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
 import _ from 'lodash'
 
 // Components
-import {ErrorHandling} from 'src/shared/decorators/errors'
-import TelegrafConfig from 'src/telegrafs/components/TelegrafConfig'
-import {
-  ComponentColor,
-  Button,
-  RemoteDataState,
-  SpinnerContainer,
-  TechnoSpinner,
-  Overlay,
-  ComponentStatus,
-} from '@influxdata/clockface'
+import {Overlay} from '@influxdata/clockface'
+import GetResources from 'src/resources/components/GetResources'
+import TelegrafConfigOverlayForm from 'src/telegrafs/components/TelegrafConfigOverlayForm'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
-// Utils
-import {downloadTextFile} from 'src/shared/utils/download'
-import {getByID} from 'src/resources/selectors'
+// Actions
+import {updateTelegraf} from 'src/telegrafs/actions/thunks'
 
 // Types
-import {AppState, Telegraf, ResourceType} from 'src/types'
+import {ResourceType, Telegraf} from 'src/types'
 
 interface OwnProps {
   onClose: () => void
 }
 
-interface StateProps {
-  telegraf: Telegraf
-  status: RemoteDataState
-  telegrafConfig: string
-  configStatus: RemoteDataState
-}
-
-type Props = OwnProps & StateProps
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = OwnProps & ReduxProps
 
-@ErrorHandling
-class TelegrafConfigOverlay extends PureComponent<Props> {
-  public render() {
-    return <>{this.overlay}</>
+const TelegrafConfigOverlay: FC<Props> = ({onUpdateTelegraf, onClose}) => {
+  const handleUpdateTelegraf = (telegraf: Telegraf): void => {
```

Comment:
Since we're just updating the telegraf and closing the overlay (and we have access to the `onClose` function in the `TelegrafConfigOverlayForm`, let's move this into that component to localize the logic. That way we can remove redux from this component altogether